oggm/core/preprocessing/climate.py

```diff
@@ -136,6 +136,230 @@ def process_custom_climate_data(gdir):
     out = {'climate_source': fpath, 'hydro_yr_0': y0+1, 'hydro_yr_1': y1}
     gdir.write_pickle(out, 'climate_info')
 
+@entity_task(log, writes=['gcm_data'])
+def process_gcm_data(gdir):
+    """Processes and writes the climate data for this glacier.
+
+    Interpolates the GCM data to the high-resolution CL2 climatologies
+    (provided with OGGM) and writes everything to a NetCDF file.
+    """
+    # read the climatology high resolution CL2 climatologies
+    clfile = utils.get_cru_cl_file()
+    ncclim = salem.GeoNetcdf(clfile)
+
+    # GCM temperature and precipitation data
+    if not (('gcm_temp_file' in cfg.PATHS) and \
+                    os.path.exists(cfg.PATHS['gcm_temp_file'])):
+        raise IOError('GCM temp file not found')
+
+    if not (('gcm_precc_file' in cfg.PATHS) and \
+                    os.path.exists(cfg.PATHS['gcm_precc_file'])):
+        raise IOError('GCM precc file not found')
+
+    if not (('gcm_precl_file' in cfg.PATHS) and \
+                    os.path.exists(cfg.PATHS['gcm_precl_file'])):
+        raise IOError('GCM precl file not found')
+
+
+    # read the file
+    fpathTemp = cfg.PATHS['gcm_temp_file']
+    fpathPrecc = cfg.PATHS['gcm_precc_file']
+    fpathPrecl = cfg.PATHS['gcm_precl_file']
+    temp = salem.open_xr_dataset(fpathTemp)
+    temp = temp.TREFHT
+    temp.time
+    temp = temp[9:-3, :, :] # from normal years to hydrological years
+    precpC = xr.open_dataset(fpathPrecc)
+    precpL = xr.open_dataset(fpathPrecl)
+    precp = precpC.PRECC+precpL.PRECL
+    precp = precp[9:-3, :, :] # from normal years to hydrological years
+
+    precp.time # Is this line needed?
+    temp['time'] = pd.period_range('850-10', '2005-9', freq='M')
+    precp['time'] = pd.period_range('850-10', '2005-9', freq='M')
```

Comment:
Here also, this decision could be based on info obtained from the file. But let's keep it like this for now.